docs/getting-started/upgrade_guide_v4.md

```diff
@@ -0,0 +1,130 @@
+
+To simplify the following step, we will be using **Linux** for illustration. Binary for
+[Mac(arm64)](https://github.com/crypto-org-chain/chain-main/releases/download/v4.2.2/chain-main_4.2.2_Darwin_arm64.tar.gz) and [Windows](https://github.com/crypto-org-chain/chain-main/releases/download/v4.2.2/chain-main_4.2.2_Windows_x86_64.zip) are also available. 
+
+- Terminate the `chain-maind`; afterwards, download the `4.2.2` released binaries from github:
+
+  ```bash
+  $ curl -LOJ https://github.com/crypto-org-chain/chain-main/releases/download/v4.2.2/chain-main_4.2.2_Linux_x86_64.tar.gz
+  $ tar -zxvf chain-main_4.2.2_Linux_x86_64.tar.gz
+  ```
+
+
+    ::: tip Remarks: 
+    If you have stated `chain-maind` with *systemd* service, kindly stop it by 
+
+    ```bash 
+    $ sudo systemctl stop chain-maind
+    ```
+    And replace the binary in the location where the `ExecStart` states in Systemd Unit file.
+    
+    :::
+
+
+### Step 1.1 -  Verify the version
+
+You can verify the installation by checking the version of `chain-maind`, the latest version is `4.2.2`.
+
+  ```bash 
+  # check the version of chain-maind
+  $ ./chain-maind version
+  4.2.2
+  ```
+
+### Step 1.2 -  Insert new configuration
+
+Several new fields under `~/.chain-maind/config/app.toml` have been introduced, kindly add the following fields to the `app.toml`:
+- Under `Base Configuration`, please the aligned the`app-db-backend` with the `db-backend` field under `~/.chain-maind/config/config.toml`, for example, if it is `goleveldb` (default), then we can add:
+```bash
+# AppDBBackend defines the database backend type to use for the application and snapshots DBs.
+# An empty string indicates that a fallback will be used.
+# First fallback is the deprecated compile-time types.DBBackend value.
+# Second fallback (if the types.DBBackend also isn't set), is the db-backend value set in Tendermint's config.toml.
+app-db-backend = "goleveldb"
+```
+ 
+
+
+- Under `Rosetta Configuration`:
+```bash
+# EnableDefaultSuggestedFee defines if the server should suggest fee by default.
+# If 'construction/medata' is called without gas limit and gas price,
+# suggested fee based on gas-to-suggest and denom-to-suggest will be given.
+enable-fee-suggestion = false
+
+# GasToSuggest defines gas limit when calculating the fee
+gas-to-suggest = 200000
+
+# DenomToSuggest defines the defult denom for fee suggestion.
+# Price must be in minimum-gas-prices.
+denom-to-suggest = "basecro"
+```
+- Under `gRPC Configuration`:
+```bash
+# MaxRecvMsgSize defines the max message size in bytes the server can receive.
+# The default value is 10MB.
+max-recv-msg-size = "10485760"
+
+# MaxSendMsgSize defines the max message size in bytes the server can send.
+# The default value is math.MaxInt32.
+max-send-msg-size = "2147483647"
+```
+
+::: tip Remarks:
+If you are upgrading from `v3.3.4` or earlier, kindly also add  the following fields to the `app.toml` under `Base Configuration`:
```

Comment:
Might also link to https://crypto.org/docs/getting-started/mainnet.html#step-0-notes-on-network-upgrades
as it contains explanation about the iavl setting?